filebeat/prospector/prospector_log.go

```diff
@@ -42,12 +43,33 @@ func NewLog(p *Prospector) (*Log, error) {
 	return prospectorer, nil
 }
 
+func (l *Log) resolvePaths() error {
+	if l.paths != nil {
```

Comment:
I would avoid making a distinction between non-nil and zero length slices. This can lead to subtle programming mistakes. I would check for `len(l.paths) > 0`